command/agent/agent_endpoint.go

```diff
@@ -136,8 +139,41 @@ func (s *HTTPServer) listServers(resp http.ResponseWriter, req *http.Request) (i
 		return nil, CodedError(501, ErrInvalidMethod)
 	}
 
-	// Get the current list of servers
-	return client.Servers(), nil
+	// Preallocate for at least 5x servers
```

Comment:
Get rid of this and guard against this being called on a node only with a server